flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -305,9 +306,52 @@ SqlCreate SqlCreateFunction(Span s, boolean replace, boolean isTemporary) :
             <PYTHON>   { functionLanguage = "PYTHON"; }
         )
     ]
+    [ <USING>  {
+        if ("SQL".equals(functionLanguage) || "PYTHON".equals(functionLanguage)) {
+            throw new ParseException(
+                String.format("USING JAR syntax is not applicable to %s language.", functionLanguage));
```

Comment:
Please use `throw SqlUtil.newContextException` to throw exception during parsing.